### DISCUSSION TITLE
MaaFramework 术语命名讨论

### DISCUSSION THREAD

**Discussion by moomiji** [MaaXYZ/MaaFramework#470] [Ideas]
### node
- 定义：符合任务流水线（Pipeline）协议的一个完整的  `JsonObject`

### task
- 定义：若干 `node` 按一定次序相连的逻辑顺序结构——从头到尾的整个过程

### entry
- 定义：一个 `task` 中的第一个 `node`

### pipeline
- 定义：一个 pipeline 文件夹中包含的 `node` 全体

### bundle
- 定义：一个按标准资源结构存储的文件夹，包含 pipeline、model、Image 文件

### resource
- 定义：多个 `bundle` 按一定次序加载的资源结构

```mermaid
stateDiagram-v2

classDef others fill:#0000,stroke:#e5e7eb

state resource {
    state bundle {
        state pipeline {
            state task {
                entry --> node
                node --> [*]
            }
            tasks
        }

        model
        image
    }
    bundles
}

tasks:::others
bundles:::others
```

**Comment by MistEO** [comment]
layer 我觉得很困惑，不直观

**Reply by Windsland52** [reply]
bundle如何

**Reply by MistEO** [reply]
来自 GPT：

> 对于这样的场景，你可以考虑用 "Asset" 这个单词。它在游戏开发、图形渲染等领域中常用于描述资源文件夹，且很容易让开发者联想到资源的加载和管理。以下是几个其他的选项：
> 
> Resource
> 这是通用的术语，直接表示“资源”。开发者很容易理解其含义。
> 
> Repository
> 表示资源的集合，有一种存储库的意味，但稍显正式。
> 
> Library
> 适合用来表达一组可加载的资源或功能模块。
> 
> Container
> 强调资源的存储与组织，但稍微有点笼统。
> 
> Bundle
> 表示一组打包的资源，特别适用于打包和分发的场景。
> 
> 综合考虑，你的需求建议用 "Asset" 或 "Resource"，简单、明了且专业。

**Reply by MistEO** [reply]
要么就叫 Bundle 好了，接口改名为 `MaaResourcePostBundle`，怎么样

**Reply by MistEO** [reply]
> 要么就叫 Bundle 好了，接口改名为 `MaaResourcePostBundle`，怎么样

后续我想着可以增加 MaaResourcePostPipeline, MaaResourcePostModel 这种更灵活的独立接口。PostBundle 也能比较好的和这些区分开

**Reply by neko-para** [reply]
> > 要么就叫 Bundle 好了，接口改名为 `MaaResourcePostBundle`，怎么样
> 
> 后续我想着可以增加 MaaResourcePostPipeline, MaaResourcePostModel 这种更灵活的独立接口。PostBundle 也能比较好的和这些区分开

我觉得得把现在啥都postxxx的格式改下（这完全就念不通啊

**Reply by MistEO** [reply]
> > > 要么就叫 Bundle 好了，接口改名为 `MaaResourcePostBundle`，怎么样
> > 
> > 
> > 后续我想着可以增加 MaaResourcePostPipeline, MaaResourcePostModel 这种更灵活的独立接口。PostBundle 也能比较好的和这些区分开
> 
> 我觉得得把现在啥都postxxx的格式改下（这完全就念不通啊

那叫啥

**Reply by neko-para** [reply]
> > > > 要么就叫 Bundle 好了，接口改名为 `MaaResourcePostBundle`，怎么样
> > > 
> > > 
> > > 后续我想着可以增加 MaaResourcePostPipeline, MaaResourcePostModel 这种更灵活的独立接口。PostBundle 也能比较好的和这些区分开
> > 
> > 我觉得得把现在啥都postxxx的格式改下（这完全就念不通啊
> 
> 那叫啥

比如AsyncLoadBundle？

**Reply by MistEO** [reply]
感觉接口叫 async 也挺怪的，async 是实现不是接口做的事情（

**Reply by neko-para** [reply]
> 感觉接口叫 async 也挺怪的，async 是实现不是接口做的事情（

不加是也可以啊, 只是可能会和sync版接口搞混, 主要是这个意思

**Reply by MistEO** [reply]
post的都是异步的（

要不在文档提一嘴

**Reply by neko-para** [reply]
> post的都是异步的（
> 
> 要不在文档提一嘴

就是说现在为了表明异步这个特点, 加了个读不通顺的post(
所以说想改改

**Comment by MistEO** [comment]
https://github.com/MaaXYZ/MaaFramework/blob/main/docs/zh_cn/1.2-术语解释.md

**Comment by MistEO** [comment]
6 什么时候还画了个图，要不要放文档里去
